src/VscProcess.cpp

```diff
@@ -176,30 +197,21 @@ void VscProcess::receivedDisplayOnCommand(const hri_safe_remote_control_system::
   // Turn on custom display mode
   vsc_send_user_feedback(vscInterface, VSC_USER_DISPLAY_MODE, DISPLAY_MODE_CUSTOM_TEXT);
 
-  // Checks if the display messages are below the MAXCHARACTERS limit
-  checkCharacterLimit(msg);
-
-  // Update Display messages
-  vsc_send_user_feedback_string(vscInterface, VSC_USER_DISPLAY_ROW_1, msg.displayrow1.c_str());
-  vsc_send_user_feedback_string(vscInterface, VSC_USER_DISPLAY_ROW_2, msg.displayrow2.c_str());
-  vsc_send_user_feedback_string(vscInterface, VSC_USER_DISPLAY_ROW_3, msg.displayrow3.c_str());
-  vsc_send_user_feedback_string(vscInterface, VSC_USER_DISPLAY_ROW_4, msg.displayrow4.c_str());
+  // Update Display message
+  vsc_send_user_feedback_string(vscInterface, VSC_USER_DISPLAY_ROW_4, msg->data.c_str());
 }
 
 void VscProcess::receivedDisplayOffCommand(const std_msgs::EmptyConstPtr& msg)
 {
-  hri_safe_remote_control_system::SrcDisplay clear_msg;
-  clear_msg.displayrow1 = clear_msg.displayrow2 = clear_msg.displayrow3 = clear_msg.displayrow4 = "";
-  
   if (vscInterface == NULL)
   {
     return;
   }
 
-  vsc_send_user_feedback_string(vscInterface, VSC_USER_DISPLAY_ROW_1, clear_msg.displayrow1.c_str());
-  vsc_send_user_feedback_string(vscInterface, VSC_USER_DISPLAY_ROW_2, clear_msg.displayrow2.c_str());
-  vsc_send_user_feedback_string(vscInterface, VSC_USER_DISPLAY_ROW_3, clear_msg.displayrow3.c_str());
-  vsc_send_user_feedback_string(vscInterface, VSC_USER_DISPLAY_ROW_4, clear_msg.displayrow4.c_str());
+  vsc_send_user_feedback_string(vscInterface, VSC_USER_DISPLAY_ROW_1, "");
```

Comment:
I'm not sure if `""` is a valid c_str. I'm at HQ so let me know if this needs to be tested.

Reply:
It will convert the literal. Should be no problem.